client/src/components/VisibilityControl/VisibilityControl.tsx

```diff
@@ -84,7 +103,7 @@ export class VisibilityControl extends React.Component {
               <div className="visibility-control__components">
                 {_.map(
                   item_component_order,
-                  (component) => item_components[component]
+                  (component: string) => item_components[component]
```

Comment:
I think if the above typing is sorted out, this should be able to figure out that it's `"count" | "icon" | "text"` automatically, won't need to be told it's a string.